sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.Response.cs

```diff
@@ -0,0 +1,103 @@
+    {
+        private sealed class HttpClientTransportResponse : Response
+        {
+            private string _clientRequestId;
+            private readonly PipelineResponse _pipelineResponse;
+
+            public HttpClientTransportResponse(string clientRequestId, PipelineResponse pipelineResponse)
+            {
+                _clientRequestId = clientRequestId;
+                _pipelineResponse = pipelineResponse;
+            }
+
+            public override int Status => _pipelineResponse.Status;
+
+            public override string ReasonPhrase => _pipelineResponse.ReasonPhrase;
+
+            public override string ClientRequestId
+            {
+                get => _clientRequestId;
+                set => _clientRequestId = value;
+            }
+
+            public override Stream? ContentStream
+            {
+                get => _pipelineResponse.ContentStream;
+                set => _pipelineResponse.ContentStream = value;
+            }
+
+            public override BinaryData Content
+            {
+                get
+                {
+                    ResetContentStreamPosition(_pipelineResponse);
+                    return _pipelineResponse.Content;
+                }
+            }
+
+            public override BinaryData ReadContent(CancellationToken cancellationToken = default)
+                => _pipelineResponse.ReadContent(cancellationToken);
+
+            public override async ValueTask<BinaryData> ReadContentAsync(CancellationToken cancellationToken = default)
+                => await base.ReadContentAsync(cancellationToken).ConfigureAwait(false);
+
+            protected internal override bool ContainsHeader(string name)
+                => _pipelineResponse.Headers.TryGetValue(name, out _);
+
+            protected internal override IEnumerable<HttpHeader> EnumerateHeaders()
+            {
+                foreach (KeyValuePair<string, string> header in _pipelineResponse.Headers)
+                {
+                    yield return new HttpHeader(header.Key, header.Value);
+                }
+            }
+
+            protected internal override bool TryGetHeader(string name, [NotNullWhen(true)] out string? value)
+                => _pipelineResponse.Headers.TryGetValue(name, out value);
+
+            protected internal override bool TryGetHeaderValues(string name, [NotNullWhen(true)] out IEnumerable<string>? values)
+                => _pipelineResponse.Headers.TryGetValues(name, out values);
+
+            public override void Dispose()
+            {
+                PipelineResponse response = _pipelineResponse;
+                ResetContentStreamPosition(response);
+                response?.Dispose();
+            }
+
+            private static void ResetContentStreamPosition(PipelineResponse response)
+            {
+                if (response.ContentStream is MemoryStream stream && response.ContentStream.CanSeek &&
```

Comment:
why is it important that this be a `MemoryStream`?

Reply:
This supports both mocks and this type's implementation of ContentStream - and prevents us from trying to reset the position of a live network stream, which would throw if we did that.  We're able to throw in ClientModel if the stream's position is non-zero, but it would be a breaking change to Azure.Core if we did it there.

Reply:
I thought a live network stream would fail the `response.ContentStream.CanSeek` evaluation. Is there a case where stream is not a `MemoryStream` but `CanSeek` is true?

Reply:
That's fair.  We should be able to do this for any stream where CanSeek is true.

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-net/pull/42294

Reply:
Actually - I was wrong.  We do need this.  Azure.Core has a `ResponseShouldNotBeUsedStream` that it replaces Response.ContentStream with when `ExtractResponseContent` is called.  It throws if you call `CanSeek` on it.  Checking for `MemoryStream` here restricts the check to either buffered content streams or content streams set by mocks in our test implementations.  This is not ideal, but I don't think we want to rework this part at the moment.  We can file an issue to revisit it if needed.